### PR TITLE
[bitnami/nginx] deployment fail because of OpenShift restricted-v2 issue 25425

### PR DESCRIPTION
**Pull request by antoinetran**
### Description of the change

Fix [[bitnami/nginx] deployment fail because of OpenShift restricted-v2](https://github.com/bitnami/charts/issues/25425#top) 
### Benefits

Deployment to OpenShit with restricted-v2 policy will not fail because of initContainer with missing podSecurityContext

### Possible drawbacks

None AFAIK.

### Applicable issues

- fixes #

### Checklist

- [X] Chart version bumped in `Chart.yaml` according to [semver](http://semver.org/). This is *not necessary* when the changes only affect README.md files.
- [X] Title of the pull request follows this pattern [bitnami/<name_of_the_chart>] Descriptive title
- [X] All commits signed off and in agreement of [Developer Certificate of Origin (DCO)](https://github.com/bitnami/charts/blob/main/CONTRIBUTING.md#sign-your-work)

### CODE REVIEW

File: bitnami/nginx/templates/deployment.yaml [hunk 1]
```diff
@@ -80,7 +80,7 @@ spec:
           image: {{ include "nginx.image" . }}
           imagePullPolicy: {{ .Values.image.pullPolicy | quote }}
           {{- if .Values.containerSecurityContext.enabled }}
-          securityContext: {{- omit .Values.containerSecurityContext "enabled" | toYaml | nindent 12 }}
+          securityContext: {{- include "common.compatibility.renderSecurityContext" (dict "secContext" .Values.podSecurityContext "context" $) | nindent 12 }}
```
Comment (jotamartos) [inline bitnami/nginx/templates/deployment.yaml:83]: You need to use this as you are configuring the security context of the container

```suggestion
          securityContext: {{- include "common.compatibility.renderSecurityContext" (dict "secContext" .Values.containerSecurityContext "context" $) | nindent 12 }}
```

**Review by jotamartos — CHANGES_REQUESTED** [review]
Thanks for your contribution. Please create separate PRs for the 2 solutions, you can still use this one for nginx but please create a separate one for nginx-ingress-controller.

**Comment by antoinetran** [comment]
> Thanks for your contribution. Please create separate PRs for the 2 solutions, you can still use this one for nginx but please create a separate one for nginx-ingress-controller.

Done, the nginx-ingress-controller commits have been reverted in this branch.

**Review by jotamartos — APPROVED** [review]
LGTM! Thank you for your contribution.

**Comment by jotamartos** [comment]
I accepted the PR, but there is a problem with the validations. Could you please follow the instructions here to solve it? 

https://github.com/bitnami/charts/pull/25443/checks?check_run_id=24677473747

**Comment by antoinetran** [comment]
> I accepted the PR, but there is a problem with the validations. Could you please follow the instructions here to solve it?
> 
> https://github.com/bitnami/charts/pull/25443/checks?check_run_id=24677473747

Ok I rebased the branch and simply removed the extra commits related to nginx-ingress-controller.

**Review by jotamartos — APPROVED** [review]

**Pull request merged**